### PR TITLE
Adding Windows Support for System Agent

### PR DESCRIPTION
**Pull request by phillipsj**
This adds the necessary support for adding Windows plan and system agent support. This adds tests to cover the required changes and removes the previous work-around we had in place to bypass the system agent checks in Rancher.

* https://github.com/rancher/windows/issues/87

Signed-off-by: Jamie Phillips <jamie.phillips@suse.com>

### CODE REVIEW

**Review by nickgerace — DISMISSED** [review]

**Comment by phillipsj** [comment]
rke2running
itjoined

File: pkg/provisioningv2/rke2/planner/planner_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+				os:              "linux",
+				command:         "sh",
+				scriptName:      "run.sh",
+				envs:            []string{"INSTALL_RKE2_EXEC"},
+			},
+		},
+		{
+			name: "Checking Windows Instructions",
+			args: args{
+				version:         "v1.21.5+rke2r2",
```
Comment (rosskirkpat) [inline pkg/provisioningv2/rke2/planner/planner_test.go:42]: Should we instead be using an rke2 version that supports the system-agent in RKE2?

Comment (phillipsj) [reply pkg/provisioningv2/rke2/planner/planner_test.go:42]: This is just a test to ensure that versions are translated correctly, the string doesn't really matter.

**Review by luthermonson — APPROVED** [review]

**Comment by phillipsj** [comment]
image
image

**Pull request merged**